pkgs/development/haskell-modules/configuration-ghc-7.10.x.nix

```diff
@@ -42,6 +42,10 @@ self: super: {
   # Build jailbreak-cabal with the latest version of Cabal.
   jailbreak-cabal = super.jailbreak-cabal.override { Cabal = self.Cabal_1_24_2_0; };
 
+  gtk2hs-buildtools = super.gtk2hs-buildtools.override { Cabal = self.Cabal_1_24_2_0; };
```

Comment:
Cabal is itself a cabal package available on hackage. GHC pins a specific version on each release and distributes it among its boot packages. gtk2hs-buildtools requires a newer major version of Cabal than the one that is pinned by GHC 7.10.x (that would be 1.22.x.x). For that reason the package fails to build at present, supplying the newer Cabal dependency fixes it.